lib/plugins/aws/invokeLocal/index.js

```diff
@@ -215,47 +225,29 @@ class AwsInvokeLocal {
   }
 
   checkDockerDaemonStatus() {
-    return new BbPromise((resolve, reject) => {
-      const docker = spawn('docker', ['version']);
-      docker.on('exit', error => {
-        if (error) {
-          reject('Please start the Docker daemon to use the invoke local Docker integration.');
-        }
-        resolve();
-      });
+    return spawnExt('docker', ['version']).catch(() => {
+      throw Error('Please start the Docker daemon to use the invoke local Docker integration.');
     });
   }
 
-  checkDockerImage() {
-    const runtime = this.getRuntime();
-
-    return new BbPromise((resolve, reject) => {
-      const docker = spawn('docker', ['images', '-q', `lambci/lambda:${runtime}`]);
-      let stdout = '';
-      docker.stdout.on('data', buf => {
-        stdout += buf.toString();
-      });
-      docker.on('exit', error => {
-        return error ? reject(error) : resolve(Boolean(stdout.trim()));
-      });
-    });
+  checkDockerImage(imageName) {
+    return spawnExt('docker', ['images', '-q', imageName]).then(
+      ({ stdoutBuffer }) => {
+        return Boolean(stdoutBuffer.toString().trim());
+      },
+      () => false
```

Comment:
In `master` exception is propagated, here we're silencing it. Is this intended? (I don't see much changes in how outcome of `checkDockerImage` is handled)